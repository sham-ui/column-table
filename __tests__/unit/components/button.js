import { onclick, disabled } from 'sham-ui-directives';
import Button  from '../../../src/components/button.sfc';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( Button, {}, {
        directives: {
            onclick,
            disabled
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'buttonText', () => {
    const meta = renderer( Button, {
        item: 'Foo',
        index: 1,
        buttonText: ( item, index ) => item + index
    }, {
        directives: {
            onclick,
            disabled
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
