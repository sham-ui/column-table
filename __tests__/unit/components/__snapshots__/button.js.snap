// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonClassNames 1`] = `
Object {
  "Options": Object {
    "buttonClassNames": [Function],
    "buttonText": [Function],
    "disabled": [Function],
    "iconClassNames": "",
    "index": 1,
    "item": "foo",
  },
  "html": "<button class=\\"foo1\\"><!--0--></button>",
}
`;

exports[`buttonText 1`] = `
Object {
  "Options": Object {
    "buttonClassNames": [Function],
    "buttonText": [Function],
    "disabled": [Function],
    "iconClassNames": "",
    "index": 1,
    "item": "Foo",
  },
  "html": "<button class=\\"\\"><!--0-->Foo1</button>",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "buttonClassNames": [Function],
    "buttonText": [Function],
    "disabled": [Function],
    "iconClassNames": "",
  },
  "html": "<button class=\\"\\"><!--0--></button>",
}
`;
